src/js/app/modules/users/controllers/editUsers.controller.js

```diff
@@ -0,0 +1,112 @@
+            // Convert date strings to Date type objects
+            vm.userInfo.birthday = new Date(vm.userInfo.birthday);
+            vm.userInfo.hiredTime = new Date(vm.userInfo.hiredTime);
+
+            vm.pageName = data.name + ' ' + data.firstLastName;
+        });
+
+        DocumentTypeFactory.query(function (data){
+            vm.documentTypeApi = data;
+        });
+        BirthPlaceFactory.query(function (data){
+            vm.birthPlaceApi = data;
+        });
+        NationalityFactory.query(function (data){
+            vm.nationalityApi = data;
+        });
+        MaritalStatusFactory.query(function (data){
+            vm.maritalStatusApi = data;
+        });
+        ScholarshipFactory.query(function (data){
+            vm.scholarshipApi = data;
+        });
+        RhFactory.query(function (data){
+            vm.rhApi = data;
+        });
+        SeniorityFactory.query(function (data){
+            vm.seniorityApi = data;
+        });
+        ProjectFactory.query(function (data){
+            vm.projectApi = data;
+        });
+        AfpFactory.query(function (data){
+            vm.afpApi = data;
+        });
+        EpsFactory.query(function (data){
+            vm.epsApi = data;
+        });
+        SkillsFactory.query(function (data){
+            vm.skillsApi = data;
+        });
+
+        vm.validateFormData = function (){
+            // TODO: Convert this to a reusable component
+            // instead of a controller method
+
+            var userForm = $scope.editUsersForm;
+
+            for (var key in userForm) {
+                var elem = userForm[key];
+
+                // Look for chaged fields
+                if (typeof(elem) == 'object' && elem.hasOwnProperty('$dirty') && elem.$dirty === true) {
+                    var modelValue = null;
+
+                    // On these fields save just the id and not the whole object
+                    switch (elem.$name) {
+                        case 'socialStratum':
+                        case 'documentType':
+                        case 'birthPlace':
+                        case 'nationality':
+                        case 'maritalStatus':
+                        case 'scholarship':
+                        case 'socialStratum':
+                        case 'rh':
+                        case 'seniority':
+                        case 'project':
+                        case 'afp':
+                        case 'eps':
+                            modelValue = elem.$modelValue.id;
+                            break;
+
+                        default:
+                            modelValue = elem.$modelValue;
+                    }
+
+                    vm.postObject[elem.$name] = modelValue;
+                }
+            }
+
+            UsersFactory.update(vm.postObject, {id: $routeParams.id}, function (response){
+                vm.postStatus = response;
+                vm.postStatus.message = "User successfully created";
```

Comment:
This should be better to see a "User successfully updated" or something similar that mentions update not creation